plantcv/plantcv/roi/roi_methods.py

```diff
@@ -194,7 +192,161 @@ def _draw_roi(img, roi_contour):
+    gm_y = GaussianMixture(n_components=nrows, random_state=0).fit(centers_y)
+    clusters_x = np.sort(gm_x.means_[:, 0])
+    clusters_y = np.sort(gm_y.means_[:, 0])
+    spacing_x = (clusters_x[ncols-1] - clusters_x[0])/(ncols-1)if ncols > 1 else 0
+    spacing_y = (clusters_y[nrows-1] - clusters_y[0])/(nrows-1) if nrows > 1 else 0
+    spacing = (round(spacing_x), round(spacing_y))
+    coord = (round(clusters_x[0]), round(clusters_y[0]))
+    return coord, spacing
+
+
+def _adjust_radius_coord(height, width, coord, radius):
+    x = [i[0] for i in coord]
+    y = [i[1] for i in coord]
+    return _adjust_radius_max_min(height, width, radius, max(x), min(x), max(y), min(y))
+
+
+def _adjust_radius_grid(height, width, coord, radius, spacing, nrows, ncols):
+    xmax = coord[0] + (ncols-1)*spacing[0]
+    xmin = coord[0]
+    ymax = coord[1] + (nrows-1)*spacing[1]
+    ymin = coord[1]
+    return _adjust_radius_max_min(height, width, radius, xmax, xmin, ymax, ymin)
+
+
+def _adjust_radius_max_min(height, width, radius, xmax, xmin, ymax, ymin):
+    if ((xmin < 0) or (xmax > width) or (ymin < 0) or (ymax > height)):
+        fatal_error("An ROI extends outside of the image!")
+    distances_to_edge = [xmin, width-xmax, ymin, height-ymax]
+    min_distance = min(distances_to_edge)
+    if min_distance < radius:
+        print('Shrinking radius to make ROIs fit in the image')
+        radius = min_distance - 1
+    return(radius)
+
+
+def _rois_from_coordinates(img, coord=None, radius=None):
+    if radius is None:
+        fatal_error("Specify a radius if creating rois from a list of coordinates")
+    # Get the height and width of the reference image
+    height, width = np.shape(img)[:2]
+    radius = _adjust_radius_coord(height, width, coord, radius)
+    overlap_img = np.zeros((height, width))
+    # Initialize a binary image of the circle that will contain all ROI
+    all_roi_img = np.zeros((height, width), dtype=np.uint8)
+    roi_contour = []
+    roi_hierarchy = []
+    for i in range(0, len(coord)):
+        # Initialize a binary image for each circle
+        bin_img = np.zeros((height, width), dtype=np.uint8)
+        y = coord[i][1]
+        x = coord[i][0]
+        # Draw the circle on the binary image
+        # Keep track of all roi
+        all_roi_img = cv2.circle(all_roi_img, (x, y), radius, 255, -1)
+        # Keep track of each roi individually to check overlapping
+        circle_img = cv2.circle(bin_img, (x, y), radius, 255, -1)
+        overlap_img = overlap_img + circle_img
+        # Make a list of contours and hierarchies
+        rc, rh = cv2.findContours(circle_img, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE)[-2:]
+        roi_contour.append(rc)
+        roi_hierarchy.append(rh)
+    return roi_contour, roi_hierarchy, overlap_img, all_roi_img
+
+
+def _grid_roi(img, nrows, ncols, coord=None, radius=None, spacing=None):
+    if radius is None:
+        if spacing[0] == 0:
+            radius = round(0.325*spacing[1])
```

Comment:
I think we should give 0.325 a name and define it in upper-case to indicate that its a constant